vernac/g_vernac.mlg

```diff
@@ -1019,6 +1019,8 @@ GRAMMAR EXTEND Gram
       | IDENT "TypeClasses" -> { PrintTypeClasses }
       | IDENT "Instances"; qid = smart_global -> { PrintInstances qid }
       | IDENT "Coercions" -> { PrintCoercions }
+      | IDENT "Notations"; qid = smart_global; l = OPT univ_name_list ->
```

Comment:
why not
```suggestion
      | IDENT "Notation"; qid = smart_global; l = OPT univ_name_list ->
```

Reply:
Originally a typo, then we were discussing which would make more sense. I don't know yet what we want to do about the fact that a given notation can have many meanings. One option would be to output information for all of them, and then I could see it being called "Print Notations". But I guess it's still one notation with many different meanings? So perhaps it should be "Print Notation" regardless.

Reply:
I would lean on `Print Notation` since plural command names are always confusing.

Reply:
Note that in terms of parsing, a notation is canonical, resolution to a particular operator goes later on. So if you only care about parsing data, there is one entry by notation.

A different thing is the algorithm used for locate, that does a very fuzzy maching on notations as to support `Locate "+"`. This is a case where the vernacular shows its limitations as a query language, and IMHO it is not worth to fix.

Reply:
There is the question of "Do we want to have multiple results?". I think it might be more prudent to have a `Print Notation` and `Location Notation` command. That way the latter can query like a locate command, but still return extra notation info.

Reply:
You could have a location command that returns the key, and a command that returns detailed info about the key, but yeah, at some point vernac as query language doesn't cut it. Maybe it would be more interesting to write a notation explorer in jsCoq?

Reply:
But each key still has many different possible interpretations, right? `"_ + _"` can be nat addition or type addition, for example.

Reply:
@ana-borges , yes, but that's after parsing; so indeed there are two levels. "_ + _" is canonical wrt parsing

Then, elaboration will make a choice. But for example, and that's the limitation custom entries wanted to fix, you can't have a `+` operator with different precedences, all of them share the same parsing rule.

Reply:
So for a particular notation key (a string like this), you have:
- parsing rules
- printing rules (which are tricky)
- its interpretation information, which is indeed a multimap to diferent regular terms.